### PR TITLE
Some code inspection fixes

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

### CODE REVIEW

File: clients/clearly-defined/src/main/kotlin/ClearlyDefinedService.kt [hunk 1]
```diff
@@ -29,7 +29,6 @@ import kotlinx.coroutines.awaitAll
 import kotlinx.coroutines.runBlocking
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
-import kotlinx.serialization.decodeFromString
```
Comment (sschuberth) [inline clients/clearly-defined/src/main/kotlin/ClearlyDefinedService.kt:32]: Hmm, interesting that these were not found by neither detekt nor Qodana. Any idea why that is, @fviernau?

File: clients/clearly-defined/src/main/kotlin/Extensions.kt [hunk 2]
```diff
@@ -19,8 +19,6 @@
 
 package org.ossreviewtoolkit.clients.clearlydefined
 
-import java.util.Locale
```
Comment (sschuberth) [inline clients/clearly-defined/src/main/kotlin/Extensions.kt:22]: Hint: It's last use was removed in 809c89d4e865e8fbd5e39b24efdaa34ad1f41c06.

File: plugins/package-managers/gradle-plugin/src/main/kotlin/GradleModelExtensions.kt [hunk 3]
```diff
@@ -51,9 +50,3 @@ internal fun Configuration.isRelevant(): Boolean {
 
     return canBeResolved && !isDeprecatedConfiguration && !isDependenciesMetadata
 }
-
-/**
- * Return whether the Android plugin for Gradle has been applied to the project.
- */
-internal fun Project.isAndroidProject(): Boolean =
```
Comment (sschuberth) [inline plugins/package-managers/gradle-plugin/src/main/kotlin/GradleModelExtensions.kt:58]: I originally wanted to keep this around as I was planning to take it into use again, but as I don't know when that will be I'm fine with removing it. I know how to restore the implementation in any case.

**Review by sschuberth — APPROVED** [review]

File: clients/clearly-defined/src/main/kotlin/ClearlyDefinedService.kt (hunk 1 above)
Comment (fviernau) [reply clients/clearly-defined/src/main/kotlin/ClearlyDefinedService.kt:32]: I don't know it for sure. But this file uses a function named `decodeFromString()` but from a different package.

**Comment by fviernau** [comment]
Merge despite failing unrelated test.

**Pull request merged**